src/nativescript-angular/common/detached-loader.ts

```diff
@@ -1,21 +1,22 @@
-import {DynamicComponentLoader, ComponentRef, ElementRef, Component, Type} from 'angular2/core';
+import {DynamicComponentLoader, ComponentRef, Component, Type, ViewChild, ViewContainerRef} from 'angular2/core';
 
 /**
- * Wrapper component used for loading compnenets when navigating
+ * Wrapper component used for loading components when navigating
  * It uses DetachedContainer as selector so that it is elementRef is not attached to the visual tree.
  */
 @Component({
     selector: 'DetachedContainer', 
     template: `<Placeholder #loader></Placeholder>`
 })
 export class DetachedLoader {
+    @ViewChild('loader', { read: ViewContainerRef }) viewContainerRef: ViewContainerRef;

```

Comment:
The fix may also be related to this. It sounds like this is the new way to get this reference, however it may need something different for the router to work.